tooling/lsp/src/requests/inlay_hint.rs

```diff
@@ -95,13 +95,21 @@ impl<'a> InlayHintCollector<'a> {
                         self.push_type_hint(lsp_location, &typ, editable);
                     }
                     ReferenceId::StructMember(struct_id, field_index) => {
-                        let struct_type = self.interner.get_struct(struct_id);
+                        let struct_type = self.interner.get_type(struct_id);
                         let struct_type = struct_type.borrow();
                         let field = struct_type.field_at(field_index);
                         self.push_type_hint(lsp_location, &field.typ, false);
                     }
+                    ReferenceId::EnumVariant(type_id, variant_index) => {
+                        let typ = self.interner.get_type(type_id);
+                        let shared_type = typ.clone();
+                        let typ = typ.borrow();
+                        let variant_type = typ.variant_function_type(variant_index, shared_type);
+                        self.push_type_hint(lsp_location, &variant_type, false);
+                    }
```

Comment:
I think the `StructMember` case is when an identifier refers to a struct field, like here:

![image](https://github.com/user-attachments/assets/4b86da15-24cd-4910-b095-932fdcf3385c)

I don't know if an identifier would point to an enum variant where that identifier would have an inlay hint. But I also think enums aren't type-checked yet so I couldn't try code with enums.

Reply:
I wasn't sure if we'd want to give an inlay hint for them either but I thought it was a possibility so I figured I might as well support it. I'd be open to removing it in the future if we don't want it.